File: Objective-C/UIKit/UIButton/UIButton+Util.m
//
//  UIButton+Util.m
//  UIButtonTest
//
//  Created by AngieMIta on 2017/8/24.
//  Copyright © 2017年 crazyit.org. All rights reserved.
//

#import "UIButton+Util.h"

@implementation UIButton (Util)

- (void)setTitleForAllState:(NSString *)title{
    [self setTitle:title forState:UIControlStateNormal];
    [self setTitle:title forState:UIControlStateHighlighted];
    [self setTitle:title forState:UIControlStateDisabled];
    [self setTitle:title forState:UIControlStateSelected];
}

- (void)setTitleColorForAllState:(UIColor *)color{
    [self setTitleColor:color forState:UIControlStateNormal];
    [self setTitleColor:color forState:UIControlStateHighlighted];
    [self setTitleColor:color forState:UIControlStateDisabled];
    [self setTitleColor:color forState:UIControlStateSelected];
}

- (void)setImageForAllState:(UIImage *)image{
    [self setImage:image forState:UIControlStateNormal];
    [self setImage:image forState:UIControlStateHighlighted];
    [self setImage:image forState:UIControlStateDisabled];
    [self setImage:image forState:UIControlStateSelected];
}


@end

File: Swift/UIButton+Util.swift
//
//  UIButton+Util.swift
//  Test
//
//  Created by 巴糖 on 2017/5/28.
//  Copyright © 2017年 巴糖. All rights reserved.
//

import UIKit

extension UIButton{
    
    var title: String {
        get {
            return self.title(for: UIControlState.normal)!
        }
        set {
            self.setTitle(newValue, for: UIControlState.normal)
            self.setTitle(newValue, for: UIControlState.selected)
            self.setTitle(newValue, for: UIControlState.highlighted)
            self.setTitle(newValue, for: UIControlState.disabled)
        }
    }
    
    var image: UIImage {
        get {
            return self.image(for: UIControlState.normal)!
        }
        set {
            self.setImage(newValue, for: UIControlState.normal)
            self.setImage(newValue, for: UIControlState.selected)
            self.setImage(newValue, for: UIControlState.highlighted)
            self.setImage(newValue, for: UIControlState.disabled)
        }
    }
    
    var backgroundImage: UIImage {
        get {
            return self.backgroundImage(for: UIControlState.normal)!
        }
        set {
            self.setBackgroundImage(newValue, for: UIControlState.normal)
            self.setBackgroundImage(newValue, for: UIControlState.selected)
            self.setBackgroundImage(newValue, for: UIControlState.highlighted)
            self.setBackgroundImage(newValue, for: UIControlState.disabled)
        }
    }
    
    var titleColor: UIColor {
        get {
            return self.titleColor(for: UIControlState.normal)!
        }
        set {
            self.setTitleColor(newValue, for: UIControlState.normal)
            self.setTitleColor(newValue, for: UIControlState.selected)
            self.setTitleColor(newValue, for: UIControlState.highlighted)
            self.setTitleColor(newValue, for: UIControlState.disabled)
        }
    }
    
    
}
